### PR TITLE
Courtney Buratto- React Todo

### PR DESCRIPTION
**Pull request by cocoitali**
@decagondev

### CODE REVIEW

File: src/components/TodoComponents/TodoForm.js [hunk 1]
```diff
@@ -2,7 +2,8 @@ import React from 'react';
 
 const TodoForm = props => {
     return (
-        <div>    
+        <div>
+            <input placeholder="...todo"></input>
```
Comment (decagondev) [inline src/components/TodoComponents/TodoForm.js]: Nice use of the spread operator!

File: src/components/TodoComponents/TodoForm.js [hunk 2]
```diff
@@ -2,12 +2,12 @@ import React from 'react';
...
 const TodoForm = props => {
     return (
-        <div>
-            <input placeholder="...todo"></input>
-            <button>Add Todo</button>
-            <button>Clear Completed</button>
-        </div>
-    )
+        <form className = "todo-form">
+            <input value ={props.value} type="text" onChange={props.changeInput} placeholder= "...todo"></input>
```
Comment (decagondev) [inline src/components/TodoComponents/TodoForm.js]: nice use of the spread operator and great work getting this functional

File: src/App.js [hunk 3]
```diff
@@ -1,13 +1,73 @@
...
+import TodoForm from "./components/TodoComponents/TodoForm";
+import TodoList from "./components/TodoComponents/TodoList";
+
 
 class App extends React.Component {
-  // you will need a place to store your state in this component.
+  constructor() {
+    super();
+    this.state = {
+        item: '', //stores values passed into input
```
Comment (decagondev) [inline src/App.js]: good use of comments here

File: src/App.js [hunk 4]
```diff
@@ -1,13 +1,73 @@
...
+          }
+        ], //stores values added to todo list
+    };
+  }
+
+  //HANDLERS
+
+  //input handler
+  onChange = (event) => {
+    this.setState({[event.target.name]: event.target.value});
```
Comment (decagondev) [inline src/App.js]: great use of multipurpose input handler great way to keep your code DRY!

File: src/App.js [hunk 5]
```diff
@@ -1,13 +1,73 @@
...
+  }
+
+  //add handler
+  onSubmit = (event) => {
```
Comment (decagondev) [inline src/App.js]: nice use of arrow function here for the handlers

File: src/App.js [hunk 6]
```diff
@@ -1,13 +1,73 @@
...
+    event.preventDefault();
+    const todos = this.state.todos.slice()
+    todos.push(
+      {task:this.state.item, completed: false, id: Date.now()});
```
Comment (decagondev) [inline src/App.js]: good use of Date.now() to set the id

File: src/App.js [hunk 7]
```diff
@@ -1,13 +1,73 @@
...
+      this.setState({todos, item: ""});
+  }    
+
+  toggleComplete = id => {
```
Comment (decagondev) [inline src/App.js]: looking great here. remember to comment a bit more to get in to good code style practice

File: src/components/TodoComponents/Todo.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+const Todo = props => {
+    return (
+        <div onClick={() => props.toggleComplete(props.id)} className="todo-display">
+            {props.task}
```
Comment (decagondev) [inline src/components/TodoComponents/Todo.js]: good use of toggle method here

File: src/components/TodoComponents/TodoForm.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+
+const TodoForm = props => {
+    return (
+        <div className = "todo-form-wrapper">
+          <input name="item" type="text" onChange={props.onChange} value={props.item} placeholder= "...todo" />
```
Comment (decagondev) [inline src/components/TodoComponents/TodoForm.js]: yet more good use of the spread operator

File: src/components/TodoComponents/TodoList.js [hunk 10]
```diff
@@ -1,2 +1,19 @@
...
+import React from 'react';
+import Todo from './Todo';
+
+
+const TodoList = props => {
+    return (
+        <div className="todo-list">
+        {props.item.map(item => {
+            return <Todo
+                key={item.id}
```
Comment (decagondev) [inline src/components/TodoComponents/TodoList.js]: good clean use of id as the key for the map

File: src/App.js (hunk 7 above)
Comment (cocoitali) [reply src/App.js]: thanks. I'll be working on making this one functional today.

File: src/components/TodoComponents/Todo.js [hunk 11]
```diff
@@ -1,12 +1,17 @@
 import React from 'react';
 
+
 const Todo = props => {
+    console.log(props.task.id)
+    // const completed = props.task.completed ? 'completed' : 'notCompleted';
```
Comment (decagondev) [inline src/components/TodoComponents/Todo.js]: great use of ternary operator for the string replacement in the class name

File: src/components/TodoComponents/TodoList.js [hunk 12]
```diff
@@ -5,15 +5,18 @@ import Todo from './Todo';
...
-        {props.item.map(item => {
+        {props.todos.map(item => {
             return <Todo
-                key={item.id}
-                id={item.id}
-                task={item.task}
-                onClick={props.toggleComplete}/>
-            })}
+                toggleCompleted={props.toggleCompleted}
+                key={item.key}
```
Comment (decagondev) [inline src/components/TodoComponents/TodoList.js]: interesting use of item.key here. could you elaborate on your choice for this :)

**Pull request closed without merging**